ccl/changefeedccl: TestChangefeedRandomExpressions failed

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/6e668f8e-d8ad-4fb7-873a-5e4ed14f0ff8?testReportRun=4&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-25.2 @ [4c66ca53ca72240e24224cd78cf4d89c3741b200](https://github.com/cockroachdb/cockroach/commits/4c66ca53ca72240e24224cd78cf4d89c3741b200):


```
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true AND st_pointinsidecircle(st_endpoint('0102000080020000003E1E127F9125F7C17977A7250A86F7C150191C07B7D3FE414C3D222CBEB3E641EBA9A565FF8E02C2A0B1856032A6BFC1':::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY, seed._float8::FLOAT8, 0.5147132257040723:::FLOAT8::FLOAT8, seed._float4::FLOAT8)::BOOL)
    changefeed_test.go:1320: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true AND st_pointinsidecircle(st_endpoint('0102000080020000003E1E127F9125F7C17977A7250A86F7C150191C07B7D3FE414C3D222CBEB3E641EBA9A565FF8E02C2A0B1856032A6BFC1':::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY, seed._float8::FLOAT8, 0.5147132257040723:::FLOAT8::FLOAT8, seed._float4::FLOAT8)::BOOL) because error pq: st_pointinsidecircle(): radius of the circle has to be positive
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed.col_37 !~ seed.col_37
    changefeed_test.go:1320: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed.col_37 !~ seed.col_37 because error pq: column "seed.col_37" does not exist
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1331: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 49 events
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT seed._bool)
    changefeed_test.go:1331: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT seed._bool) to emit 44 events
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT similar_to_escape(seed._string::STRING, seed._string::STRING, to_char(NULL::TIMESTAMPTZ, seed._string::STRING)::STRING::STRING)::BOOL)
    changefeed_test.go:1327: Skipping predicate (NOT similar_to_escape(seed._string::STRING, seed._string::STRING, to_char(NULL::TIMESTAMPTZ, seed._string::STRING)::STRING::STRING)::BOOL) because it returned no rows
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (_st_dwithin(st_geogfromgeojson(seed._string::STRING)::GEOGRAPHY::GEOGRAPHY, st_s2covering(NULL::GEOGRAPHY)::GEOGRAPHY::GEOGRAPHY, seed._float4::FLOAT8, seed._bool::BOOL)::BOOL AND EXISTS (SELECT seed.tableoid AS col_100, seed.tableoid AS col_101 FROM d.public.seed AS "ta'b%v_27" WHERE seed._bool))
    changefeed_test.go:1327: Skipping predicate (_st_dwithin(st_geogfromgeojson(seed._string::STRING)::GEOGRAPHY::GEOGRAPHY, st_s2covering(NULL::GEOGRAPHY)::GEOGRAPHY::GEOGRAPHY, seed._float4::FLOAT8, seed._bool::BOOL)::BOOL AND EXISTS (SELECT seed.tableoid AS col_100, seed.tableoid AS col_101 FROM d.public.seed AS "ta'b%v_27" WHERE seed._bool)) because it returned no rows
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1327: Skipping predicate false because it returned no rows
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT '$':::STRING AS col_146, NULL AS "c\fol_147" FROM d.public.seed AS tab_45 JOIN d.public.seed AS "ta b_46" ON (tab_45.crdb_internal_mvcc_timestamp) = ("ta b_46"._decimal) JOIN d.public.seed AS "tab _47" ON (("ta b_46"._int4) = ("tab _47"._int4)) AND ((tab_45.tableoid) = ("tab _47".tableoid)) WHERE true ORDER BY seed._inet, seed._bytes ASC NULLS LAST, seed._enum DESC, seed._int4)
    changefeed_test.go:1327: Skipping predicate EXISTS (SELECT '$':::STRING AS col_146, NULL AS "c\fol_147" FROM d.public.seed AS tab_45 JOIN d.public.seed AS "ta b_46" ON (tab_45.crdb_internal_mvcc_timestamp) = ("ta b_46"._decimal) JOIN d.public.seed AS "tab _47" ON (("ta b_46"._int4) = ("tab _47"._int4)) AND ((tab_45.tableoid) = ("tab _47".tableoid)) WHERE true ORDER BY seed._inet, seed._bytes ASC NULLS LAST, seed._enum DESC, seed._int4) because it returned no rows
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1331: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (((st_disjoint('0104000080070000000101000080BCE4E53298CDD9C134772C624B26F2C1AC9812C96661F3410101000080A367375D9BECF0C1206A4CCA9F06B5C11C56A38C0119D6C1010100008010BE69ABAB2001C2B287E42F10BDE5C1F8BAC2C6FAFEFD4101010000803F4C31C82EEDF3C13AABD2EE1DF2EAC1A0B6AEDB286BC4C101010000809C696207DC8CE6418F430F42C863F2C1BA7EC32DD895E1C101010000808825FBED5D48FF41082D5AA30B62D241B02AB6662A9FD9C10101000080FD7166F95F8F00C238485F10BA06FF416AF97F4EF2EFF8C1':::GEOMETRY::GEOMETRY, '0101000000000000000000F87F000000000000F87F':::GEOMETRY::GEOMETRY)::BOOL AND seed._bool) OR (seed._inet::INET IN (seed._inet, '20.143.9.22/24':::INET, '6174:ed34:f5b8:2568:3805:176f:c5f9:6717/57':::INET, seed._inet))) OR false)
    changefeed_test.go:1320: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (((st_disjoint('0104000080070000000101000080BCE4E53298CDD9C134772C624B26F2C1AC9812C96661F3410101000080A367375D9BECF0C1206A4CCA9F06B5C11C56A38C0119D6C1010100008010BE69ABAB2001C2B287E42F10BDE5C1F8BAC2C6FAFEFD4101010000803F4C31C82EEDF3C13AABD2EE1DF2EAC1A0B6AEDB286BC4C101010000809C696207DC8CE6418F430F42C863F2C1BA7EC32DD895E1C101010000808825FBED5D48FF41082D5AA30B62D241B02AB6662A9FD9C10101000080FD7166F95F8F00C238485F10BA06FF416AF97F4EF2EFF8C1':::GEOMETRY::GEOMETRY, '0101000000000000000000F87F000000000000F87F':::GEOMETRY::GEOMETRY)::BOOL AND seed._bool) OR (seed._inet::INET IN (seed._inet, '20.143.9.22/24':::INET, '6174:ed34:f5b8:2568:3805:176f:c5f9:6717/57':::INET, seed._inet))) OR false) because error pq: st_disjoint(): geos: no locations to init GEOS
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._timestamptz::TIMESTAMPTZ IN (SELECT COALESCE(seed._timestamptz, seed._timestamptz)::TIMESTAMPTZ AS "col}_162" FROM d.public.seed AS tab_57)) AND seed._bool)
    changefeed_test.go:1331: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((seed._timestamptz::TIMESTAMPTZ IN (SELECT COALESCE(seed._timestamptz, seed._timestamptz)::TIMESTAMPTZ AS "col}_162" FROM d.public.seed AS tab_57)) AND seed._bool) to emit 47 events
    changefeed_test.go:1334: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._int8::INT8 IN (SELECT ((-7943651158531917051):::INT8::INT8 | NULL::INT8)::INT8::INT8 AS col_170 FROM d.public.seed AS tab_66 JOIN d.public.seed AS "ta%vb_67" JOIN d.public.seed AS tab_68 ON (("ta%vb_67"._int8) = (tab_68._int2)) AND (("ta%vb_67".crdb_internal_mvcc_timestamp) = (tab_68.crdb_internal_mvcc_timestamp)) ON ((tab_66._bytes) = ("ta%vb_67"._bytes)) AND ((tab_66._int8) = ("ta%vb_67"._int2)) WHERE (seed._bool OR true) GROUP BY seed._enum, seed._float8 ORDER BY seed._enum DESC NULLS LAST, seed._enum DESC NULLS FIRST)
    changefeed_test.go:1327: Skipping predicate seed._int8::INT8 IN (SELECT ((-7943651158531917051):::INT8::INT8 | NULL::INT8)::INT8::INT8 AS col_170 FROM d.public.seed AS tab_66 JOIN d.public.seed AS "ta%vb_67" JOIN d.public.seed AS tab_68 ON (("ta%vb_67"._int8) = (tab_68._int2)) AND (("ta%vb_67".crdb_internal_mvcc_timestamp) = (tab_68.crdb_internal_mvcc_timestamp)) ON ((tab_66._bytes) = ("ta%vb_67"._bytes)) AND ((tab_66._int8) = ("ta%vb_67"._int2)) WHERE (seed._bool OR true) GROUP BY seed._enum, seed._float8 ORDER BY seed._enum DESC NULLS LAST, seed._enum DESC NULLS FIRST) because it returned no rows
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT 'H':::STRING:::NAME AS col_175 FROM d.public.seed AS "tab
        _71" JOIN d.public.seed AS ta😈b_72 ON ((("tab
        _71"._float4) = (ta😈b_72._float4)) AND (("tab
        _71".crdb_internal_mvcc_timestamp) = (ta😈b_72._decimal))) AND (("tab
        _71"._bool) = (ta😈b_72._bool)) GROUP BY seed._timestamptz, seed._float8, seed._decimal HAVING (seed._bool OR (seed._string ~* seed._string)) ORDER BY seed._timestamptz ASC NULLS LAST, seed._decimal DESC NULLS LAST)
    changefeed_test.go:1327: Skipping predicate EXISTS (SELECT 'H':::STRING:::NAME AS col_175 FROM d.public.seed AS "tab
        _71" JOIN d.public.seed AS ta😈b_72 ON ((("tab
        _71"._float4) = (ta😈b_72._float4)) AND (("tab
        _71".crdb_internal_mvcc_timestamp) = (ta😈b_72._decimal))) AND (("tab
        _71"._bool) = (ta😈b_72._bool)) GROUP BY seed._timestamptz, seed._float8, seed._decimal HAVING (seed._bool OR (seed._string ~* seed._string)) ORDER BY seed._timestamptz ASC NULLS LAST, seed._decimal DESC NULLS LAST) because it returned no rows
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string::STRING NOT IN (SELECT e'S\x10z\x168*U\r':::STRING::STRING AS col_180 FROM d.public.seed AS tab_75 JOIN d.public.seed AS "t%pab_76" JOIN d.public.seed AS tab_77 ON ("t%pab_76"._int8) = (tab_77._int8) ON ((tab_75._int2) = ("t%pab_76"._int8)) AND ((tab_75._float8) = (tab_77._float8)))
    changefeed_test.go:1331: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._string::STRING NOT IN (SELECT e'S\x10z\x168*U\r':::STRING::STRING AS col_180 FROM d.public.seed AS tab_75 JOIN d.public.seed AS "t%pab_76" JOIN d.public.seed AS tab_77 ON ("t%pab_76"._int8) = (tab_77._int8) ON ((tab_75._int2) = ("t%pab_76"._int8)) AND ((tab_75._float8) = (tab_77._float8))) to emit 93 events
    changefeed_test.go:1334: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL::GEOMETRY NOT IN ('010500008009000000010200008003000000F0661E72906ECAC130708708138ED6C11081771F8435D4C18F1EE293EE77F0C14AE10896DAA6F9C17C29107A50A9FC41D60C2BC8A6B80142741176F2705BDFC1AE98BFC2B22201420102000080060000009EEA5DC186C8F941E3951057CDD9F3C1F43884DC507DF141067C990E301BF841149DBBB4A294E2C1225AC05A2F8CEDC1B083C8100EC7FB41C82DBE20B553EA41C097884CF8F2FEC1C8614D718D45D241A43010463212E941B090C2281436C741D8996CFC6798D3C1D28DB21CB60C02422A9F0F8521F4ECC16AC4B41DD5EFEEC1C0D3429484F2D941080A0BC60039DD410102000080040000005875CE751601FC41BCE425FB02EFDCC1001E0F787151BD4128D86E455FD5FB417CBE8B680453E141F09771EB571500C2E829416546CFFB416E0796B0B94EF9410CAF5727CA2ADDC138AF5F4E524EE741B60D06B4F800F841A08E023D5D64E741010200008004000000A8849CC06F7A01428290C68AC92B0142885E0B7B388EFC41A629F95751AFF241502236F9A601E941F8B4BB5AA7F4FF4130577AFD7AD8D94124769856A287E941540BF2074EF5EE41460C97C5D7CEFFC1683BAE5FD536D44194D58265A908FD41010200008002000000CBC72942D237F4C15018E5947D3CF9C1D4B03C50C13FFC411E374E641B06EAC1C35E63670D19F1C11C51ABD556A3EE4101020000800300000048182B6099EAD041860D5C5FC6E4F241A85E31E9E019D1C1C48E2A18275FDFC17076F786DF0BF241087FE371FA19DC414CDCC784B0BB01C230AB756B0123EE41B2968E16B510FFC1010200008006000000AC4A4A99170BF7419A103369AD95E6C1E00F02F8270CF9C1D0E19FA7A4E2F941B0FD98937EF2DFC1A4837FEC826DEB4174185C586958EF41B850315D0DB8DB41489F66905B32D9C16615E4D29276F0410854B2EADE32F541AE57662D53C3E1C1F020887C2782E34128AD49E853BBF441246A2C85EFB1E24100296B51F4AF814198EFFC86C063F741C8E98BDB4D85DEC1010200008003000000DEE83300BC05F541C0C6D15FB91CDDC148EAB3648B9CEF410EC704089069F641D824B41A8816E941A25EF242B9A10142B83AC2558198D64178185C65738402420AA0BD99B94701C2010200008007000000DA672A4BE10600C256CE31F80CA0F0C1788D97AADF53F24157BB374D0EA7F6C1EB0FCE196BA7F2C1007DCBD0A5B8DA4188E1D498A878E8C1A0C633B23F8F01C2A0F88702222AFE41EA2A55FF28C0E7C19ED654EC8745F6C106AED4D0E0ADE0C170C90F480554CC412E78369294ADF4C1261557BD55F0F041A032652BF09BE0C100E1EDF1F9909A4109699E3920D300C2404BEE379158DC410E62D71A87C601424084D3DA8D7FA041':::GEOMETRY,)
    changefeed_test.go:1327: Skipping predicate NULL::GEOMETRY NOT IN ('010500008009000000010200008003000000F0661E72906ECAC130708708138ED6C11081771F8435D4C18F1EE293EE77F0C14AE10896DAA6F9C17C29107A50A9FC41D60C2BC8A6B80142741176F2705BDFC1AE98BFC2B22201420102000080060000009EEA5DC186C8F941E3951057CDD9F3C1F43884DC507DF141067C990E301BF841149DBBB4A294E2C1225AC05A2F8CEDC1B083C8100EC7FB41C82DBE20B553EA41C097884CF8F2FEC1C8614D718D45D241A43010463212E941B090C2281436C741D8996CFC6798D3C1D28DB21CB60C02422A9F0F8521F4ECC16AC4B41DD5EFEEC1C0D3429484F2D941080A0BC60039DD410102000080040000005875CE751601FC41BCE425FB02EFDCC1001E0F787151BD4128D86E455FD5FB417CBE8B680453E141F09771EB571500C2E829416546CFFB416E0796B0B94EF9410CAF5727CA2ADDC138AF5F4E524EE741B60D06B4F800F841A08E023D5D64E741010200008004000000A8849CC06F7A01428290C68AC92B0142885E0B7B388EFC41A629F95751AFF241502236F9A601E941F8B4BB5AA7F4FF4130577AFD7AD8D94124769856A287E941540BF2074EF5EE41460C97C5D7CEFFC1683BAE5FD536D44194D58265A908FD41010200008002000000CBC72942D237F4C15018E5947D3CF9C1D4B03C50C13FFC411E374E641B06EAC1C35E63670D19F1C11C51ABD556A3EE4101020000800300000048182B6099EAD041860D5C5FC6E4F241A85E31E9E019D1C1C48E2A18275FDFC17076F786DF0BF241087FE371FA19DC414CDCC784B0BB01C230AB756B0123EE41B2968E16B510FFC1010200008006000000AC4A4A99170BF7419A103369AD95E6C1E00F02F8270CF9C1D0E19FA7A4E2F941B0FD98937EF2DFC1A4837FEC826DEB4174185C586958EF41B850315D0DB8DB41489F66905B32D9C16615E4D29276F0410854B2EADE32F541AE57662D53C3E1C1F020887C2782E34128AD49E853BBF441246A2C85EFB1E24100296B51F4AF814198EFFC86C063F741C8E98BDB4D85DEC1010200008003000000DEE83300BC05F541C0C6D15FB91CDDC148EAB3648B9CEF410EC704089069F641D824B41A8816E941A25EF242B9A10142B83AC2558198D64178185C65738402420AA0BD99B94701C2010200008007000000DA672A4BE10600C256CE31F80CA0F0C1788D97AADF53F24157BB374D0EA7F6C1EB0FCE196BA7F2C1007DCBD0A5B8DA4188E1D498A878E8C1A0C633B23F8F01C2A0F88702222AFE41EA2A55FF28C0E7C19ED654EC8745F6C106AED4D0E0ADE0C170C90F480554CC412E78369294ADF4C1261557BD55F0F041A032652BF09BE0C100E1EDF1F9909A4109699E3920D300C2404BEE379158DC410E62D71A87C601424084D3DA8D7FA041':::GEOMETRY,) because it returned no rows
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT '64 years 1 mon 180 days 07:23:22.705547':::INTERVAL AS c̃ol_246, '010500000002000000010200000003000000000000000000244000000000000024400000000000003440000000000000344000000000000024400000000000004440010200000004000000000000000000444000000000000044400000000000003E400000000000003E40000000000000444000000000000034400000000000003E400000000000002440':::GEOMETRY AS "col _247" FROM d.public.seed AS tab_108 WHERE seed._bool GROUP BY seed._int8 HAVING (((NOT seed._bool) OR seed._bool) OR seed._bool) ORDER BY seed._int8)
    changefeed_test.go:1331: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT '64 years 1 mon 180 days 07:23:22.705547':::INTERVAL AS c̃ol_246, '010500000002000000010200000003000000000000000000244000000000000024400000000000003440000000000000344000000000000024400000000000004440010200000004000000000000000000444000000000000044400000000000003E400000000000003E40000000000000444000000000000034400000000000003E400000000000002440':::GEOMETRY AS "col _247" FROM d.public.seed AS tab_108 WHERE seed._bool GROUP BY seed._int8 HAVING (((NOT seed._bool) OR seed._bool) OR seed._bool) ORDER BY seed._int8) to emit 49 events
    changefeed_test.go:1334: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._enum::greeting IN (SELECT seed._enum::greeting AS col_263 FROM d.public.seed AS tab_115 JOIN d.public.seed AS "Tab_116" ON ((tab_115._int8) = ("Tab_116"._int8)) AND ((tab_115._date) = ("Tab_116"._date)) JOIN (VALUES (seed._int4), ((-16796):::INT8), ((-25545):::INT8), (NULL)) AS tab_117 (col_262) ON ((tab_115._int8) = (tab_117.col_262)) AND (("Tab_116"._int8) = (tab_117.col_262)) WHERE false ORDER BY seed.col_262 ASC NULLS FIRST)
    changefeed_test.go:1320: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._enum::greeting IN (SELECT seed._enum::greeting AS col_263 FROM d.public.seed AS tab_115 JOIN d.public.seed AS "Tab_116" ON ((tab_115._int8) = ("Tab_116"._int8)) AND ((tab_115._date) = ("Tab_116"._date)) JOIN (VALUES (seed._int4), ((-16796):::INT8), ((-25545):::INT8), (NULL)) AS tab_117 (col_262) ON ((tab_115._int8) = (tab_117.col_262)) AND (("Tab_116"._int8) = (tab_117.col_262)) WHERE false ORDER BY seed.col_262 ASC NULLS FIRST) because error pq: column "seed.col_262" does not exist
    changefeed_test.go:1315: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (seed._bool OR jsonb_path_exists(seed._jsonb::JSONB, '$."rXWqq"."VSCF"."fnpWNKeyycHwcU"[*]."TKenTcjNYDYQyBbhQTQ"[*]."sgNoATDXJXbDMbNmtwiE"."QhSWDjTmGKAnDx"."JlFLJtEImotMV"."dOBjlQnQt"."MCTdNmutjYlAyWJjb"."hpdxAZeacNBVXGeGX"."lKaVIfylSw"."AioxIbAlfBCoxAyKI"."RYxpt"."ELurNlEDxAqtMQ"."IfJGFFOKfk"':::JSONPATH::JSONPATH, seed._jsonb::JSONB)::BOOL))
    changefeed_test.go:1331: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND (seed._bool OR jsonb_path_exists(seed._jsonb::JSONB, '$."rXWqq"."VSCF"."fnpWNKeyycHwcU"[*]."TKenTcjNYDYQyBbhQTQ"[*]."sgNoATDXJXbDMbNmtwiE"."QhSWDjTmGKAnDx"."JlFLJtEImotMV"."dOBjlQnQt"."MCTdNmutjYlAyWJjb"."hpdxAZeacNBVXGeGX"."lKaVIfylSw"."AioxIbAlfBCoxAyKI"."RYxpt"."ELurNlEDxAqtMQ"."IfJGFFOKfk"':::JSONPATH::JSONPATH, seed._jsonb::JSONB)::BOOL)) to emit 49 events
    changefeed_test.go:1382: jsonb_path_exists(): "vars" argument is not an object
--- FAIL: TestChangefeedRandomExpressions/kafka (4.47s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #147573 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.1.7-rc]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -1362,19 +1362,22 @@ func TestChangefeedRandomExpressions(t *testing.T) {
 				// error, we unfortunately don't have the pgcode and have to
 				// rely on known strings.
 				validPgErrs := []string{
+					"argument is not an object",
 					"cannot subtract infinite dates",
-					"regexp compilation failed",
-					"invalid regular expression",
-					"invalid escape string",
-					"error parsing GeoJSON",
+					"dwithin distance cannot be less than zero",
 					"error parsing EWKB",
 					"error parsing EWKT",
-					"geometry type is unsupported",
-					"should be of length",
-					"dwithin distance cannot be less than zero",
-					"parameter has to be of type Point",
+					"error parsing GeoJSON",
 					"expected LineString",
+					"geometry type is unsupported",
+					"invalid escape string",
+					"invalid regular expression",
 					"no locations to init GEOS",
+					"parameter has to be of type Point",
+					"regexp compilation failed",
+					"result out of range",
+					"should be of length",
+					"unknown DateStyle parameter",
 				}
 				containsKnownPgErr := func(e error) (interface{}, bool) {
 					for _, v := range validPgErrs {

